doc/architecture/007-No-coupling-with-NixOS.md

```diff
@@ -0,0 +1,19 @@
+# No coupling with NixOS
+
+## Context
+
+NixOps 1 and 2 were tightly coupled with NixOS. This made it hard to use NixOps for other purposes, such as deploying Docker containers, or nix-darwin.
+
+Furthermore, it had a few special files and expectations in the `nixpkgs` repository, which made it hard for NixOS maintainers to understand and maintain, and also made it hard for NixOps to change, as NixOps is expected to support a wide range of NixOS versions.
```

Comment:
Can you link to an example here? That would help me picture what problem you're trying to solve.